source/common/upstream/load_balancer_impl.cc

```diff
@@ -116,17 +146,8 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority,
   }
 
   size_t total_load = 100;
-  int32_t first_healthy_priority = -1;
-  for (size_t i = 0; i < per_priority_health.size(); ++i) {
-    if (first_healthy_priority < 0 && per_priority_health[i] > 0) {
-      first_healthy_priority = i;
-    }
-    // Now assign as much load as possible to the high priority levels and cease assigning load
-    // when total_load runs out.
-    per_priority_load[i] =
-        std::min<uint32_t>(total_load, per_priority_health[i] * 100 / normalized_total_health);
-    total_load -= per_priority_load[i];
-  }
+  const auto first_healthy_priority =
+      distributeLoad(per_priority_load, per_priority_health, total_load, normalized_total_health);
```

Comment:
mildly weird that we return first_healthy_priority but tweak total_load as a member variable, but I'm going say it's commented heavily enough I don't actually care - more thoughts for future PRs.